src/backend/distributed/metadata/node_metadata.c

```diff
@@ -73,7 +73,7 @@ char *CurrentCluster = "default";
 bool ReplicateReferenceTablesOnActivate = true;
 
 /* did current transaction modify pg_dist_node? */
-bool TransactionModifiedNodeMetadata = false;
+bool volatile TransactionModifiedNodeMetadata = false;
```

Comment:
I cannot follow why we need `volatile` for a global flag that is only modified by transaction callbacks and a UDF. If we don't have strong reasons, I think it could be confusing to have these qualifiers.

As far as I remember, volatile is useful for (a) variables that are changed via interrupt handlers such as `QueryCancelPending `  (b) multi-threaded programs -- doesn't apply to PG (c) local variables for pg_try/pg_catch due to sigsetjmp (d) hardware interrupts -- doesn't apply to pg

Why do you think it is necessary here? And, also why only for `TransactionModifiedNodeMetadata` but not many other flags?